### PR TITLE
[script.module.inputstreamhelper] 0.4.0

### PR DESCRIPTION
**Pull request by dagwieers**
### Description
This release includes various reliability improvements and bugfixes,
a new API for add-ons and a settings menu for troubleshooting.

This release is the first release with integration tests and test
coverage, and was a collaborative effort of a handful of people.

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

### CODE REVIEW

**Comment by dagwieers** [comment]
We would like to see this release as soon as possible, since it is blocking the release of other add-ons.
And the new functionality to reinstall/remove Widevine is very welcome for troubleshooting playback issues.

File: script.module.inputstreamhelper/addon.xml [hunk 1]
```diff
@@ -1,25 +1,45 @@
...
+        <import addon="script.module.inputstreamhelper" optional="true"/>
         <import addon="script.module.kodi-six" version="0.0.4"/>
-        <import addon="script.module.requests" version="2.9.1"/>
+        <import addon="script.module.pysocks" version="1.6.8" optional="true"/>
+        <import addon="script.module.routing" version="0.2.0"/>
     </requires>
-    <extension library="lib" point="xbmc.python.module"/>
+    <extension point="xbmc.python.module" library="lib"/>
+    <!-- This is needed for running from RunPlugin() -->
+    <extension point="xbmc.python.pluginsource" library="addon.py"/>
```
Comment (enen92) [inline script.module.inputstreamhelper/addon.xml]: ?

File: script.module.inputstreamhelper/changelog.txt [hunk 2]
```diff
@@ -1,56 +1,83 @@
...
-+ Automatic Widevine CDM download on ARM devices
-+ Display Widevine EULA during installation procedure
-+ German translation (thanks to asciidisco)
-+ New, smaller and less ugly generic icon
-+ Better exception handling
-+ Code cleanup
-
-2017.09.13 v0.1.0
-+ Initial release
+## v0.4.0 (2019-09-01)
```
Comment (enen92) [inline script.module.inputstreamhelper/changelog.txt]: Changelog is no longer parsed by kodi, I'd be nice to not include it in the submissions

Comment (dagwieers) [reply script.module.inputstreamhelper/changelog.txt]: Ok, I can remove it.

IMO it is useful to have it included on the system.

File: script.module.inputstreamhelper/addon.xml (hunk 1 above)
Comment (dagwieers) [reply script.module.inputstreamhelper/addon.xml]: I tried various things to make this work. Have an API for add-ons as well as an icon that can bring users to the settings of the plugin. This is what eventually happened to work.

Comment (enen92) [reply script.module.inputstreamhelper/addon.xml]: Why not turn the navigation into a dialog select? Using this add-on as a plugin doesn't really result into a seamless user experience

Comment (dagwieers) [reply script.module.inputstreamhelper/addon.xml]: I don't know what you mean with "turn the navigation into a dialog select".
What we have now is a vast improvement of what was before and it works as designed.

Comment (enen92) [reply script.module.inputstreamhelper/addon.xml]: I just realized you don't even have any navigation inside the plugin entrypoint. Plugins are VFS add-ons that display directory items to users and provide content to Kodi. In this case your plugin add-on entrypoint is a script and you are only using the plugin structure to allow external calls to some of the methods. I am afraid you'll have to do things properly and not abuse the plugin interface. Turn add-on.py into a script and use RunScript instead.

Comment (dagwieers) [reply script.module.inputstreamhelper/addon.xml]: No, we don't want to have a myriad of scripts as interface to other add-ons.
https://github.com/emilsvennesson/script.module.inputstreamhelper/wiki/API

Comment (enen92) [reply script.module.inputstreamhelper/addon.xml]: Nop sorry this is not a plugin and you can't use it like one. If you want an external API to other scripts the add-on already has a script module entrypoint, so other add-ons import it as a dependency and call into the appropriate methods. If you need to expose that as built-in functions make addon.py a script and not a plugin.

Read the difference between plugins and scripts here: https://kodi.wiki/view/About_Add-ons

**Comment by dagwieers** [comment]
@emilsvennesson Back to you.

File: script.module.inputstreamhelper/addon.xml (hunk 1 above)
Comment (enen92) [reply script.module.inputstreamhelper/addon.xml]: Furthermore you don't need a myriad of scripts as interface to other asdons...add-on py is enough.
You parse the args just like a plugin and redirect to the appropriate method. You call it from other add-ons as `RunScript(script, args)`.

**Pull request closed without merging**